### PR TITLE
Small refactoring

### PR DESCRIPTION
**Pull request by flosse**
@birkenfeld I couldn't find a difference between `DomainHandle` and `DomainIndex` so I simplified some lines by getting rid of `DomainHandle`.

### CODE REVIEW

File: src/master.rs [hunk 1]
```diff
@@ -148,11 +150,25 @@ impl Master {
...
+            scan_busy,
+            app_time,
+            ..
+        } = data;
+        let link_up = devices
+            .iter()
+            .nth(0)
+            .ok_or_else(|| Error::new(ErrorKind::InvalidData, "No devices available"))?
+            .link_state
+            != 0;
```
Comment (birkenfeld) [inline src/master.rs]: This rustfmt-ism is not very readable to me.  What about `let first_dev = ...?; let link_up = first_dev.link_state != 0`?


Also, `iter().nth(0)` -> `get(0)`?

File: src/master.rs [hunk 2]
```diff
@@ -540,15 +557,24 @@ impl<'m> SlaveConfig<'m> {
...
+            index: *self,
+            master,
+        }
+    }
+}
+
 impl<'m> Domain<'m> {
     pub fn size(&self) -> Result<usize> {
-        ioctl!(self.master, ec::ioctl::DOMAIN_SIZE, self.index as c_ulong).map(|v| v as usize)
+        ioctl!(self.master, ec::ioctl::DOMAIN_SIZE, self.index.into()).map(|v| v as usize)
```
Comment (birkenfeld) [inline src/master.rs]: here you use `into`...

File: src/master.rs [hunk 3]
```diff
@@ -563,12 +589,17 @@ impl<'m> Domain<'m> {
         ioctl!(
             self.master,
             ec::ioctl::DOMAIN_PROCESS,
-            self.index as c_ulong
+            c_ulong::from(self.index)
```
Comment (birkenfeld) [inline src/master.rs:595]: ... and here `from`. Intentional?

File: src/types.rs [hunk 4]
```diff
@@ -3,18 +3,47 @@
...
+
+impl From<DomainIndex> for u32 {
+    fn from(idx: DomainIndex) -> Self {
+        idx.0 as u32
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct DomainDataPlacement {
+    pub offset: usize,
```
Comment (birkenfeld) [inline src/types.rs:36]: Should this be `DomainOffset` and `DomainSize` types? They don't seem to be used otherwise.

**Review by birkenfeld — APPROVED** [review]
Just a few nits.

**Comment by birkenfeld** [comment]
@flosse I think the original intent was to have `DomainHandle` be an opaque API type, while `DomainIndex` would be internal only.  As you say, it might be moot to distinguish.

File: src/types.rs (hunk 4 above)
Comment (flosse) [reply src/types.rs:36]: Yes, I combined the two values in one struct.

**Comment by flosse** [comment]
Thanks for your review!

**Comment by birkenfeld** [comment]
Good to go!

**Pull request merged**